### PR TITLE
feat:(service-worker): new service worker compatible SDK with samples

### PR DESCRIPTION
**Pull request by bardisg**
## Description of the change

Addition of a service worker compatible version based on the NodeJS SDK that can be used in a service worker.
Samples for using both the JS SDK and the service worker SDK in a Chrome extension with manifest v3

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

[> Fix [#1]() ](https://www.notion.so/rudderstacks/Use-JS-SDK-in-service-worker-93f577b3618a4ffbb4c591c1e207965f)

### Development

- [ ] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [x] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/668)

### CODE REVIEW

File: service-worker/index.js [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+      lMessage.userId = JSON.stringify(lMessage.userId);
+    }
+
+    this.queue.push({ message: lMessage, callback });
+
+    if (!this.flushed) {
+      this.flushed = true;
+      this.flush();
+      return;
+    }
```
Comment (MoumitaM) [inline service-worker/index.js:297]: This logic flushes the first event immediately.  Is this needed here?

File: tests/chrome-extension/background-script/service-worker.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+console.log(Analytics);
+
+const rudderClient = new Analytics(
+    "2EzJEim6arwPqWRy0dEp3ts1Czk",
+    "https://hosted-dev-dataplane.dev-rudder.rudderlabs.com/v1/batch",
+    {
+        logLevel: 'silly'
+    }
+);
```
Comment (MoumitaM) [inline tests/chrome-extension/background-script/service-worker.js:18]: Let's put dummy values for the write key and data plane URL.

Comment (bardisg) [reply tests/chrome-extension/background-script/service-worker.js]: fixed

File: package.json [hunk 3]
```diff
@@ -51,11 +57,15 @@
     "is": "^3.3.0",
     "lodash": "^4.17.21",
     "lodash.clonedeep": "^4.5.0",
+    "lodash.isstring": "^4.0.1",
     "lodash.merge": "^4.6.2",
     "md5": "^2.3.0",
+    "ms": "^2.1.3",
     "obj-case": "^0.2.1",
     "on-body": "^0.0.1",
+    "remove-trailing-slash": "^0.1.1",
```
[the remark this answers is not in the document]
Comment (bardisg) [reply package.json:66]: fixed

**Review by MoumitaM — APPROVED** [review]

**Pull request merged**